syntax = "proto3";

service RouteGuide {
  rpc GetFeature(coords) returns (stream ShipData) {}

}

enum Alignment {
  ALIGN_CENTER = 0;
  ALIGN_TOP = 1;
  ALIGN_BOTTOM = 2;
}

enum ShipClass {
  CORVETTE = 0;
  FRIGATE = 1;
  CRUISER = 2;
  DESTROYER = 3;
  CARRIER = 4;
  DREADNOUGHT = 5;
}

message Officer {
  string firstname = 1;
  string lastname = 2;
  string rank = 3;
}

message ShipData {
  Alignment alignment = 1;
  string name = 2;
  ShipClass ship_class = 3;
  float length = 4;
  uint32 size = 5;
  bool armed = 6;
  repeated Officer officer = 7;
  string status = 8;
}

message coords {
  string coords = 1;
}
